3-UsingAPIs/Week1/homework/ex2-checkDoubleDigits.js

```diff
@@ -12,8 +12,20 @@ Complete the function called `checkDoubleDigits` such that:
   "Expected a double digit number but got `number`", where `number` is the 
   number that was passed as an argument.
 ------------------------------------------------------------------------------*/
-function checkDoubleDigits(/* TODO add parameter(s) here */) {
-  // TODO complete this function
+function checkDoubleDigits(number) {
+  return new Promise((resolve, reject) => {
+    // check if the user passed a number
+    if (typeof number === 'number') {
+      // check the number
+      if (number >= 10 && number <= 99) {
+        resolve('This is a double digit number!');
+      } else {
+        reject(new Error(`Expected a double digit number but got ${number}`));
+      }
+    } else {
+      reject(new Error(`${number} is not a number`));
+    }
```

Comment:
Excellent explanatory comments and extra check for the type of the number. You can reduce the nesting a bit by "returning early" after detecting that the argument is not a number:

```suggestion
    if (typeof number !== 'number') {
      reject(new Error(`${number} is not a number`));
      return;
    }
    
    // check the number
    if (number >= 10 && number <= 99) {
      resolve('This is a double digit number!');
    } else {
      reject(new Error(`Expected a double digit number but got ${number}`));
    }
```

Reply:
I need to remember using return more often this way